tests/python/test_household.py

```diff
@@ -0,0 +1,377 @@
+            "label": "Test",
+            # Missing required 'data' field
+        }
+
+        response = rest_client.post(
+            "/us/household",
+            json=invalid_payload,
+            content_type="application/json",
+        )
+
+        assert response.status_code == 400
+        assert b"Missing required keys" in response.data
+
+    def test_create_household_invalid_label(self, rest_client):
+        """Test creating a household with invalid label type."""
+        invalid_payload = {
+            "data": {},
+            "label": 123,  # Should be string or None
+        }
+
+        response = rest_client.post(
+            "/us/household",
+            json=invalid_payload,
+            content_type="application/json",
+        )
+
+        assert response.status_code == 400
+        assert b"Label must be a string or None" in response.data
+
+
+class TestUpdateHousehold:
+    def test_update_household_success(
+        self, rest_client, mock_database, mock_hash_object
+    ):
+        """Test successfully updating an existing household."""
+        # Mock getting existing household
+        mock_row = MagicMock(spec=LegacyRow)
+        mock_row.__getitem__.side_effect = lambda x: SAMPLE_DB_ROW[x]
+        mock_row.keys.return_value = SAMPLE_DB_ROW.keys()
+        mock_database.query().fetchone.return_value = mock_row
+
+        updated_data = {
+            "data": {"people": {"person1": {"age": 31, "income": 55000}}},
+            "label": "Updated Test Household",
+        }
+
+        response = rest_client.put(
+            "/us/household/1",
+            json=updated_data,
+            content_type="application/json",
+        )
+        data = json.loads(response.data)
+
+        assert response.status_code == 200
+        assert data["status"] == "ok"
+        assert data["result"]["household_id"] == "1"
+
+    def test_update_nonexistent_household(self, rest_client, mock_database):
+        """Test updating a non-existent household."""
+        mock_database.query().fetchone.return_value = None
+
+        response = rest_client.put(
+            "/us/household/999",
+            json=SAMPLE_HOUSEHOLD_DATA,
+            content_type="application/json",
+        )
+        data = json.loads(response.data)
+
+        assert response.status_code == 404
+        assert data["status"] == "error"
+        assert "not found" in data["message"]
+
+    def test_update_household_invalid_payload(self, rest_client):
+        """Test updating a household with invalid payload."""
+        invalid_payload = {
+            "label": "Test",
+            # Missing required 'data' field
+        }
+
+        response = rest_client.put(
+            "/us/household/1",
+            json=invalid_payload,
+            content_type="application/json",
+        )
+
+        assert response.status_code == 400
+        assert b"Missing required keys" in response.data
+
+
+# Service level tests
+class TestHouseholdService:
```

Comment:
**question/nitt/nonblocking** 


The nitt - For the most part in most code bases there is one file per class under test. Usually it's under a directory matching the class file.

I would for that reason expect this to live in something like tests/.../services/test_household_service.py

I personally find this convention a bit confusing because I need more context to know where to look to find the appropriate test.

The question - do you currently have a convention this fits neatly into and/or do Python projects do that differently?

Reply:
I have a PR open to begin to migrate to a convention, because I agree, this method is basically a free-for-all. That said, looking forward to discussing it with you Wednesday.